Feature/#416 core notification documentation (#1120)
Avaiga/taipy-doc: docs/userman/scenario_features/events/examples.md

Examples of using Taipy event notifications to capture and consume *events*.

# Real-Time GUI Updates with Taipy Event Consumers

This example is provided to demonstrate the practical application of event-driven programming in
a real-world scenario. By capturing and processing events, developers can create responsive and
dynamic systems that notify users of important changes, such as the creation of new scenarios or
updates to data nodes. This approach enhances user experience by ensuring that users are always

> informed of relevant updates in real-time.

Reviewer: ```suggestion
This example demonstrates the power of event-driven programming in real-world applications. By capturing and processing events, developers can build responsive, dynamic systems that notify users of important changes (such as new scenario creations or data node updates) as they happen. This approach significantly enhances the user experience by providing real-time updates and ensuring users are always informed through an interactive and engaging interface.
```

This script defines a custom event consumer class `SpecificCoreConsumer`, which listens
for all events published by Taipy Core and triggers GUI notification based on those events.
It includes determining if the event is published from a `Scenario^` entity or `DataNode^` entity

> and if the action is `CREATION^` or `UPDATE^`.

Reviewer: CREATION and UPDATE are not classes or functions they are Enum values.
We can not use cross ref for those.
```suggestion
and if the action is `CREATION` or `UPDATE`.
```

!!! example
    ```python linenums="1"
    {%
    include-markdown "./code-example/user-changes-notifier.py"
    comments=false
     %}
    ```

>     This snippet shows a how you can capture and process events to notify user whenever

Reviewer: ```suggestion
    This snippet shows how you can capture and process events to notify users whenever
```

>     a new scenario is created or the value of a data node is updated.

Reviewer: ```suggestion
    a new scenario is created, or a data node's value is updated.
```

    For more details, see the [registration](understanding-topics.md) page.

# External API triggered with Taipy Event Consumers

This example illustrates how to effectively utilize event-driven programming to monitor and
respond to changes of specific event type within Taipy Core. By defining a custom event consumer,
`JobFailureCoreConsumer`, developers can seamlessly integrate external API calls triggered
by specific job status updates. This approach ensures that critical job status changes are promptly

> communicated to external systems, enhancing the robustness and responsiveness of the application.

Reviewer: ```suggestion
This example illustrates leveraging event-driven programming to monitor and respond to specific event types. By implementing a custom event consumer, `JobFailureCoreConsumer`, developers can easily trigger external API calls based on specific job status updates. This approach ensures that critical job status changes are promptly communicated to external systems, enhancing the application's monitoring and integration with third-party systems.
```

This script defines a custom event consumer class `JobFailureCoreConsumer`, which listens
for all events published by Taipy Core, when a `JOB^` entity's `status` attribute is `UPDATE`,
and triggers an external API call based on the `JOB^`'s `id`.

!!! example
    ```python linenums="1"
    {%
    include-markdown "./code-example/external-api-call-notifier.py"
    comments=false
     %}
    ```

>     This snippet shows a how you can capture and process `JOB^` events when an `UPDATE` is made to the `status`

Reviewer: ```suggestion
    This snippet shows how you can capture and process `JOB` events when an `UPDATE` is made to the `status`
```

>     of the `JOB^` and request an external API.

Reviewer: ```suggestion
    of the `JOB` and request an external API.
```

!!! example
    ```python linenums="1"
    {%
    include-markdown "./code-example/user-changes-notifier.py"
    comments=false
     %}
    ```

    This snippet shows a how you can capture and process events to notify user whenever
    a new scenario is created or the value of a data node is updated.
    For more details, see the [registration](understanding-topics.md) page.

# External API triggered with Taipy Event Consumers

This example illustrates how to effectively utilize event-driven programming to monitor and
respond to changes of specific event type within Taipy Core. By defining a custom event consumer,
`JobFailureCoreConsumer`, developers can seamlessly integrate external API calls triggered
by specific job status updates. This approach ensures that critical job status changes are promptly
communicated to external systems, enhancing the robustness and responsiveness of the application.

This script defines a custom event consumer class `JobFailureCoreConsumer`, which listens
for all events published by Taipy Core, when a `JOB^` entity's `status` attribute is `UPDATE`,
and triggers an external API call based on the `JOB^`'s `id`.

> 

Reviewer: ```suggestion
```

!!! example
    ```python linenums="1"
    {%
    include-markdown "./code-example/external-api-call-notifier.py"
    comments=false
     %}

>     ```

Reviewer: ```suggestion
```

    This snippet shows a how you can capture and process `JOB^` events when an `UPDATE` is made to the `status`
    of the `JOB^` and request an external API.

>     For more details, see the [registration](understanding-topics.md) page.

Reviewer: ```suggestion
```

Examples of using Taipy event notifications to capture and consume *events*.

# Real-Time GUI Updates with Taipy Event Consumers

> This example demonstrates the power of event-driven programming in real-world applications. By capturing and processing events, developers can build responsive, dynamic systems that notify users of important changes (such as new scenario creations or data node updates) as they happen. This approach significantly enhances the user experience by providing real-time updates and ensuring users are always informed through an interactive and engaging interface.

Reviewer: ```suggestion
This example demonstrates the power of event-driven programming in real-world applications. By
capturing and processing events, developers can build responsive, dynamic systems that notify
users of important changes (such as new scenario creations or data node updates) as they happen.
This approach significantly enhances the user experience by providing real-time updates and
ensuring users are always informed through an interactive and engaging interface.
```

This script defines a custom event consumer class `SpecificCoreConsumer`, which listens
for all events published by Taipy Core and triggers GUI notification based on those events.
It includes determining if the event is published from a `Scenario^` entity or `DataNode^` entity
and if the action is `CREATION` or `UPDATE`.

!!! example
    ```python linenums="1"
    {%
    include-markdown "./code-example/user-changes-notifier.py"
    comments=false
     %}
    ```

    This snippet shows how you can capture and process events to notify users whenever
    a new scenario is created, or a data node's value is updated.
    For more details, see the [registration](understanding-topics.md) page.

# External API triggered with Taipy Event Consumers

> This example illustrates leveraging event-driven programming to monitor and respond to specific event types. By implementing a custom event consumer, `JobFailureCoreConsumer`, developers can easily trigger external API calls based on specific job status updates. This approach ensures that critical job status changes are promptly communicated to external systems, enhancing the application's monitoring and integration with third-party systems.

Reviewer: ```suggestion
This example illustrates leveraging event-driven programming to monitor and respond to
specific event types. By implementing a custom event consumer, `JobFailureCoreConsumer`,
developers can easily trigger external API calls based on specific job status updates. This
approach ensures that critical job status changes are promptly communicated to external systems,
enhancing the application's monitoring and integration with third-party systems.
```

This script defines a custom event consumer class `JobFailureCoreConsumer`, which listens

> for all events published by Taipy Core, when a `JOB^` entity's `status` attribute is `UPDATE`,
> and triggers an external API call based on the `JOB^`'s `id`.

Reviewer: ```suggestion
for all events published by Taipy Core, when a `JOB` entity's `status` attribute is `UPDATE`,
and triggers an external API call based on the `JOB`'s `id`.
```

Examples of using Taipy event notifications to capture and consume *events*.

# Real-Time GUI Updates with Taipy Event Consumers

This example demonstrates the power of event-driven programming in real-world applications. By
capturing and processing events, developers can build responsive, dynamic systems that notify
users of important changes (such as new scenario creations or data node updates) as they happen.
This approach significantly enhances the user experience by providing real-time updates and
ensuring users are always informed through an interactive and engaging interface.

This script defines a custom event consumer class `SpecificCoreConsumer`, which listens

> for all events published by Taipy Core and triggers GUI notification based on those events.

Reviewer: ```suggestion
for all events published by Taipy and triggers GUI notification based on those events.
```

It includes determining if the event is published from a `Scenario^` entity or `DataNode^` entity
and if the action is `CREATION` or `UPDATE`.

!!! example
    ```python linenums="1"
    {%
    include-markdown "./code-example/user-changes-notifier.py"
    comments=false
     %}
    ```

    This snippet shows how you can capture and process events to notify users whenever
    a new scenario is created, or a data node's value is updated.
    For more details, see the [registration](understanding-topics.md) page.

# External API triggered with Taipy Event Consumers

This example illustrates leveraging event-driven programming to monitor and respond to
specific event types. By implementing a custom event consumer, `JobFailureCoreConsumer`,
developers can easily trigger external API calls based on specific job status updates. This
approach ensures that critical job status changes are promptly communicated to external systems,
enhancing the application's monitoring and integration with third-party systems.

This script defines a custom event consumer class `JobFailureCoreConsumer`, which listens

> for all events published by Taipy Core, when a `JOB` entity's `status` attribute is `UPDATE`,

Reviewer: ```suggestion
for all events published by Taipy, when a `JOB` entity's `status` attribute is `UPDATE`,
```
